XMonad/Util/EZConfig.hs

```diff
@@ -106,6 +107,37 @@ additionalKeysP :: XConfig l -> [(String, X ())] -> XConfig l
 additionalKeysP conf keyList =
     conf { keys = \cnf -> M.union (mkKeymap cnf keyList) (keys conf cnf) }
 
+-- |
+-- Remap keybindings from one binding to another.  For precisely, the
+-- input list contains pairs of the form @(TO, FROM)@, and maps the
+-- action bound to @FROM@ to the key @TO@.  For example, the following
+-- would bind @"M-m"@ to what's bound to @"M-c"@ (which is to close the
+-- focused window, in this case):
+--
+-- > main :: IO ()
+-- > main = xmonad $ def `remapKeysP` [("M-m", "M-c")]
+--
+-- NOTE: Submaps are not transparent, and thus these keys can't be
+-- accessed in this way: more explicitly, the @FROM@ string may **not**
+-- be a submap.  However, the @TO@ can be a submap without problems.
+-- This means that
+--
+-- > xmonad $ def `remapKeysP` [("M-m", "M-c a")]
+--
+-- is illegal (and indeed will bind @"M-m"@ to @"M-c"@!), while
```

Comment:
Should this perhaps catch and report an error (presumably to `stderr` or via `xmessage`) instead of doing something unexpected?

Reply:
We probably should, but it's harder than it looks.  We can't really use `xmessage` and friends because `remapKeysP` is not in `X` (and using `error` and such would obviously result in runtime crashes).  This is why, for the rest of EZConfig, the hack with `checkKeymap` exists.  I don't think it's worth having a whole second `checkRemapKeymap` for this, tbh, so I'm not sure what to do here.

Reply:
Better to ignore it entirely, then, instead of applying only the first part of the key sequence.

Reply:
Good point, these keybindings should now just be ignored

Reply:
Actually, it occurs to me that `@TO@` has submap issues as well: since we can't look inside a submap, if you had two bindings attempting to use the same submap (including one in `RemapP` and one in `additionalKeysP`), one would be overwritten.

Reply:
I think this is consistent with how `additionalKeysP` "allows" you to do this (rather, it doens't—maybe can't—have a mechanism to give you a compile-time error).  Plus, we also can't check whether other keybindigs are already taken, so disallowing `TO` from being a submap would seem a bit arbitrary to me

Reply:
It wasn't a problem with `additionalKeysP` because it builds the submaps first, so there's no overwriting unless you use it twice for some reason. With `remapKeysP` you now have two ways to get submaps that aren't aware of each other.

Not sure how to fix this.

Reply:
I think the end result is the same—keys are overwritten (which is why `checkKeymap` checks for exactly this).  I don't think it's a big problem, tbh, so I'd just leave it as-is, unless you have strong opinions about this